src/components/Hoverable/index.js

```diff
@@ -14,25 +14,61 @@ class Hoverable extends Component {
         this.state = {
             isHovered: false,
         };
-        this.toggleHoverState = this.toggleHoverState.bind(this);
+
+        this.wrapperView = null;
+
+        this.setIsHovered = this.setIsHovered.bind(this);
+        this.setIsNotHovered = this.setIsNotHovered.bind(this);
+        this.handleOutsideClick = this.handleOutsideClick.bind(this);
+    }
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleOutsideClick);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleOutsideClick);
+    }
+
+    /**
+     * Sets the hover state of this component to true and execute the onHoverIn callback.
+     */
+    setIsHovered() {
+        if (!this.state.isHovered) {
+            this.setState({isHovered: true}, this.props.onHoverIn);
+        }
     }
 
     /**
-     * Toggles the hover state of this component and executes the callback provided in props for that state transition.
+     * Sets the hover state of this component to false and executes the onHoverOut callback.
      */
-    toggleHoverState() {
+    setIsNotHovered() {
```

Comment:
I've made this change, although my original intent was to avoid having to use arrow functions in a callback as a performance optimization.

Reply:
Because IIRC there's some slight performance advantage to binding named functions instead of using anonymous arrow functions. Something about not having to recreate the callback functions every time the component is rendered I think?

Reply:
That's true. But it's only a performance improvement if the component you are wrapping is expensive to render. In some cases the performance advantages will be more than slight and can be extremely important to avoid doing.